cvat/apps/dataset_manager/tests/test_formats.py

```diff
@@ -484,6 +576,498 @@ def test_frames_outside_are_not_generated(self):
             self.assertTrue(frame.frame in range(6, 10))
         self.assertEqual(i + 1, 4)
 
+    def test_api_v1_tasks_annotations_dump_and_upload_with_datumaro(self):
+        test_name = self._testMethodName
+        # get formats
+        dump_formats = dm.views.get_export_formats()
+
+        for dump_format in dump_formats:
+            if dump_format.ENABLED:
+                dump_format_name = dump_format.DISPLAY_NAME
+
+                # TODO skip failed formats
+                if dump_format_name in [
+                    "CamVid 1.0", # issue #2840 and changed points values
+                    "CVAT for video 1.1", # issue #2923 and #2924
+                    "MOT 1.1", # issue #2925
+                    "MOTS PNG 1.0", # issue #2925 and changed points values
+                    "Datumaro 1.0", # not uploaded
+                    "Segmentation mask 1.1", # changed points values
+                    "WiderFace 1.0", # issue #XXX
+                ]:
+                    continue
+
+                print("dump_format_name:", dump_format_name)
+
+                # create task
+                images = self._generate_task_images(3)
+                task = self._create_task(tasks["main"], images)
+
+                # create annotations
+                if dump_format_name in ["MOT 1.1", "MOTS PNG 1.0", "PASCAL VOC 1.1", "Segmentation mask 1.1", "TFRecord 1.0", "YOLO 1.1", "ImageNet 1.0", "WiderFace 1.0", "VGGFace2 1.0"]:
+                    self._create_annotations(task, dump_format_name, "dafault")
+                else:
+                    self._create_annotations(task, dump_format_name, "random")
+
+                task_id = task["id"]
+                task_ann = TaskAnnotation(task_id)
+                task_ann.init_from_db()
+                task_data = TaskData(task_ann.ir_data, Task.objects.get(pk=task_id))
+                extractor = CvatTaskDataExtractor(task_data)
+                data_from_task_before_upload = datumaro.components.project.Dataset.from_extractors(extractor)
+
+                # dump annotations
+                url = f"/api/v1/tasks/{task_id}/annotations"
+                data = {
+                    "format": dump_format_name,
+                    "action": "download",
+                }
+                file_zip_name = osp.join(osp.dirname(__file__), 'assets', f'{test_name}_{dump_format_name}.zip')
+                self._download_file(url, data, self.admin, file_zip_name)
```

Comment:
Try to split and simplify tests as much as possible. For example, I think, we could split format tests and API tests.

Reply:
Moved tests to other file.